libexec/scoop-list.ps1

```diff
@@ -15,37 +15,44 @@ $local = installed_apps $false | ForEach-Object { @{ name = $_ } }
 $global = installed_apps $true | ForEach-Object { @{ name = $_; global = $true } }
 
 $apps = @($local) + @($global)
+if (-not $apps) {
+    Write-Host "There aren't any apps installed."
+    exit 1
+}
 
-if($apps) {
-    write-host "Installed apps$(if($query) { `" matching '$query'`"}): `n"
-    $apps | Sort-Object { $_.name } | Where-Object { !$query -or ($_.name -match $query) } | ForEach-Object {
-        $app = $_.name
-        $global = $_.global
-        $ver = Select-CurrentVersion -AppName $app -Global:$global
-
-        $install_info = install_info $app $ver $global
-        write-host "  $app " -NoNewline
-        write-host -f DarkCyan $ver -NoNewline
-
-        if($global) { write-host -f DarkGreen ' *global*' -NoNewline }
-
-        if (!$install_info) { Write-Host ' *failed*' -ForegroundColor DarkRed -NoNewline }
-        if ($install_info.hold) { Write-Host ' *hold*' -ForegroundColor DarkMagenta -NoNewline }
+$list = @()
+Write-Host "Installed apps$(if($query) { `" matching '$query'`"}):"
+$apps | Where-Object { !$query -or ($_.name -match $query) } | ForEach-Object {
+    $app = $_.name
+    $global = $_.global
+    $item = [ordered]@{}
+    $ver = Select-CurrentVersion -AppName $app -Global:$global
+    $item.Name = $app
+    $item.Version = $ver
+
+    $install_info_path = "$(versiondir $app $ver $global)\install.json"
+    $install_info = $null
+    if(Test-Path $install_info_path) {
+        $install_info = parse_json $install_info_path
+    }
 
-        if ($install_info.bucket) {
-            write-host -f Yellow " [$($install_info.bucket)]" -NoNewline
-        } elseif ($install_info.url) {
-            write-host -f Yellow " [$($install_info.url)]" -NoNewline
-        }
+    $item.Source = if ($install_info.bucket) {
+        $install_info.bucket
+    } elseif ($install_info.url) {
+        $install_info.url
+    }
 
-        if ($install_info.architecture -and $def_arch -ne $install_info.architecture) {
-            write-host -f DarkRed " {$($install_info.architecture)}" -NoNewline
-        }
-        write-host ''
+    $info = @()
+    if($global) { $info += "Global install" }
+    if (!$install_info) { $info += "Install failed" }
+    if ($install_info.hold) { $info += "Held package" }
+    if ($install_info.architecture -and $def_arch -ne $install_info.architecture) {
+        $info += $install_info.architecture
     }
-    write-host ''
-    exit 0
-} else {
-    write-host "There aren't any apps installed."
-    exit 1
+    $item.Info = $info -join ', '
+
+    $list += $item
 }
+
+$list.ForEach({[PSCustomObject]$_})
```

Comment:
```suggestion
$list.ForEach({ [PSCustomObject]$_ }) | Out-Host
```

This could solve the problem.

Reply:
Putting `Out-Host` disables the manipulation of objects in the pipeline, so this will just make it same as `Write-Host`

Reply:
`Write-Host` cannot produce correct output...

You could modify cmd shim as:

```cmd
if %errorlevel% equ 0 (
    pwsh -noprofile -ex unrestricted -command "(& 'D:\Scoop\apps\scoop\current\bin\scoop.ps1'  %args%) | Out-Host;exit $Global:LASTEXITCODE"
) else (
    powershell -noprofile -ex unrestricted -command "(& 'D:\Scoop\apps\scoop\current\bin\scoop.ps1'  %args%) | Out-Host;exit $Global:LASTEXITCODE"
)
```

Pipeline is fine in PS and bash:

![image](https://user-images.githubusercontent.com/5832170/152804415-5e0ab89a-a96c-4dc3-8534-33eb14485a51.png)

![image](https://user-images.githubusercontent.com/5832170/152804381-0268aa78-1b08-4c07-8548-17aaa21f039c.png)

I'm not familiar with CMD, you could test it yourself.

Reply:
No, fish shell and starship ^_^

Reply:
Oh, Fish shell is quite nice (I saw syntax highlighting, and you said bash, so I got confused 😅)

Reply:
Hmm, strictly, `sh`? Haha

Now `Zsh` is the default shell of Ubuntu, but it needs too many custom configurations, so I prefer Fish shell with a more simpler `oh-my-fish` (`oh-my-zsh` is too complex)

Reply:
> Zsh is the default shell of Ubuntu

Oh, I didn't know that! When did they change it?

I don't use WSL much (I use Zsh, Bash, Fish all on MSYS2 and stay on Windows :)).

Reply:
Forgotten... When I install Ubuntu using WSL2 in Windows 11 last time (mid-2021), it prompts that zsh is the default shell now...

Thanks to Windows Terminal and starship, I often forget what I'm using since basic commands (`cd`, `ls`, `rm`, etc.) and the prompt have similar interface.